test/e2e/storage/non_graceful_node_shutdown_scale.go

```diff
@@ -0,0 +1,257 @@
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2enode "k8s.io/kubernetes/test/e2e/framework/node"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+	e2eskipper "k8s.io/kubernetes/test/e2e/framework/skipper"
+	"k8s.io/kubernetes/test/e2e/storage/drivers"
+	storageframework "k8s.io/kubernetes/test/e2e/storage/framework"
+	"k8s.io/kubernetes/test/e2e/storage/utils"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+	admissionapi "k8s.io/pod-security-admission/api"
+)
+
+/*
+The Goal of this test is to do stress testing on NodeOutOfServiceVolumeDetach feature
+
+This test assumes the following:
+- The infra is GCP.
+- Number of nodes in the Kubernetes cluster is at least 10, otherwise the test will be skipped.
+- NodeOutOfServiceVolumeDetach feature is enabled.
+
+This test performs the following:
+ 1. Deploys a gce-pd csi driver.
+ 2. Creates a gce-pd csi storage class.
+ 3. Taints 50% of nodes out of 'N' nodes.
+ 4. Creates a stateful set with "N/2" number of replicas.
+ 5. Un-taint the nodes that was done at step 3.
+ 6. Shutdowns the kubelet of node on which the STS app pods is scheduled.
+    This shutdown is a non-graceful shutdown as by default the grace period is 0 on Kubelet.
+    Also, this test ensures that at least 50% of nodes are untouched so that the pods
+    can successfully get scheduled to these untouched(healthy) nodes.
+    To achieve this, step 3 is performed.
+ 7. Adds `out-of-service` taint on the node which is shut down.
+ 8. Verifies that pod gets immediately scheduled to other different healthy nodes and gets into running and ready state.
+ 9. Starts the kubelet back on the other nodes.
+ 10. Removes the `out-of-service` taint from the node.
+*/
+const (
+	// MinNodeRequired is the number of nodes required in the Kubernetes
+	// cluster in order for this test to run.
+	MinNodeRequired = 10
+)
+
+var _ = utils.SIGDescribe("[Feature:NodeOutOfServiceVolumeDetach] [Scalability] [Disruptive] [LinuxOnly] NonGracefulNodeShutdown", func() {
+	var (
+		c             clientset.Interface
+		ns            string
+		nodeListCache *v1.NodeList
+	)
+	f := framework.NewDefaultFramework("non-graceful-shutdown")
+	f.NamespacePodSecurityEnforceLevel = admissionapi.LevelPrivileged
+
+	ginkgo.BeforeEach(func(ctx context.Context) {
+		c = f.ClientSet
+		ns = f.Namespace.Name
+		e2eskipper.SkipUnlessProviderIs("gce")
+		var err error
+		nodeListCache, err = e2enode.GetReadySchedulableNodes(ctx, c)
+		if err != nil {
+			framework.Logf("Failed to list node: %v", err)
+		}
+		if len(nodeListCache.Items) < MinNodeRequired {
+			ginkgo.Skip("At least 2 nodes are required to run the test")
+		}
+	})
+
+	ginkgo.Describe("[NonGracefulNodeShutdown] pod that uses a persistent volume via gce pd driver", func() {
```

Comment:
This looks like a disruptive test and if ran in parallel with other tests could cause problem for other tests.

Reply:
Will mark the test disruptive.

Reply:
At line no 71, the top level describe is marked `Disruptive`
```
 utils.SIGDescribe("[Feature:NodeOutOfServiceVolumeDetach] [Scalability] [Disruptive] [LinuxOnly] NonGracefulNodeShutdown", func() {
 ```
 Do we still need to explicitly put `Disruptive` in here?